packages/dashboard/src/components/task-summary-panel.tsx

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import Debug from 'debug';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import { IconButton, makeStyles, Typography } from '@material-ui/core';
+import { TreeItem, TreeView } from '@material-ui/lab';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { TaskSummaryPanelItem } from './task-summary-panel-item';
+import PlayCircleFilledWhiteIcon from '@material-ui/icons/PlayCircleFilledWhite';
+import PauseCircleFilledIcon from '@material-ui/icons/PauseCircleFilled';
+import { colorPalette } from '../util/css-utils';
+import { TreeButtonGroup } from './tree-button-group';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+
+const debug = Debug('OmniPanel:TaskSummaryPanel');
+
+// TODO: this is a hacky solution to get the actor from the task status,
+// this should be changed when then backend sends an actor on the
+// taskSummary message (we should use the actor provided by the message).
+// https://github.com/osrf/rmf_core/issues/205
+export const getActorFromStatus = (status: string) => {
+  // Gets the name of the robot if it has any
+  // eslint-disable-next-line
+  return status.match(/\[[A-Za-z]([a-zA-Z0-9\/]){3,}\]+/gi);
+};
+
+export interface TaskSummaryPanelProps {
+  allTasks: RomiCore.TaskSummary[];
+}
+
+export const TaskSummaryPanel = React.memo((props: TaskSummaryPanelProps) => {
+  debug('task summary status panel render');
+
+  const { allTasks } = props;
+  const classes = useStyles();
+
+  const [taskContents, setTaskContents] = React.useState<{
+    [key: string]: JSX.Element;
+  }>({});
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+  const [selected, setSelected] = React.useState<string>('');
+
+  // We need to persist across the renders
+  const savedTasksContent = React.useRef<{
+    [key: string]: JSX.Element;
+  }>({});
+
+  const handleToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string) => {
+    setSelected(nodeIds);
+  };
+
+  const handleResetTasks = () => {
+    setExpanded([]);
+    setSelected('');
+  };
+
+  const handleClearAllCurrTasks = () => {
+    savedTasksContent.current = Object.assign({}, taskContents);
+    setTaskContents({});
+  };
+
+  const handleRestoreTasks = () => {
+    // Adding this because the test interpreter cannot find the reference to `savedTasksContent current`
+    // inside the setTaskContents
+    const storedTasks = savedTasksContent.current;
+    setTaskContents((currentContent) => {
+      // Assigning another reference.
+      let taskContentsTemp = Object.assign({}, currentContent);
```

Comment:
use `const`. Also let's rename this to `newTaskContents` to better reflect its purpose.